lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -1,49 +1,75 @@
 define((require, exports, module) ->
-    exports.setupPreviewer = (editor) ->
-        require("ace/ext/jquery")
-        katex = null
-
-        initKaTeX = (onLoaded) ->
-            # Adding KaTeX CSS
-            cssKatexPath = require.toUrl("../katex/katex.min.css")
-            linkKatex = $("<link>").attr("rel", "stylesheet").attr("href", cssKatexPath)
-            $("head").append(linkKatex)
-
-            # Adding CSS for demo formula
-            cssDemoPath = require.toUrl("./katex-demo.css")
-            linkDemo = $("<link>").attr("rel", "stylesheet").attr("href", cssDemoPath)
-            $("head").append(linkDemo)
-
-            # Adding DOM element to place formula into
-            span = $("<span>").attr("id", "formula")
-            $("body").append(span)
-
-            require(["ace/ext/katex/katex"], (katexInner) ->
-                katex = katexInner
-                onLoaded()
-                return
-            )
-            return
-
-        onLoaded = ->
-            selectedText = editor.getSelectedText()
-            try
-                katex.render(selectedText, $("#formula")[0])
-            catch e
-                $("#formula").text(e.message)
-            return
-
-        callback = (editor) ->
-            unless katex?
-                initKaTeX(onLoaded)
-                return
-            onLoaded()
-
-        editor.commands.addCommand({
-            name: "previewLaTeXFormula",
-            bindKey: {win: "Alt-p", mac: "Alt-p"},
-            exec: callback
-        })
+  exports.setupPreviewer = (editor) ->
+    katex = null
+
+    initKaTeX = (onLoaded) ->
+      # Adding CSS for demo formula
+      cssDemoPath = require.toUrl("./katex-demo.css")
+      linkDemo = $("<link>").attr(
+        rel: "stylesheet"
+        href: cssDemoPath
+      )
+      $("head").append(linkDemo)
+
+      # Adding DOM element to place formula into
+      a = $("<a>").attr(
+        href: "#"
+        id: "formula"
+        "data-toggle": "popover"
+      )
+      $("body").append(a)
+
+      require(["ace/ext/katex"], (katexInner) ->
+        katex = katexInner
+        onLoaded()
         return
+      )
+      return
+
+    options = {
+      html: true
+      placement: "bottom"
+      trigger: "manual"
+      title: "Formula"
+      content: -> katex.renderToString(editor.getSelectedText(), {displayMode: true})
+      container: "#editor"
+    }
+
+    onLoaded = ->
+      try
+        popoverPosition = $("textarea.ace_text-input").position()
```

Comment:
Can you please introduce interface for actually showing the popover, e.g. like 
```
function(jqPopoverContainer, options)
```

move this code to the implementation and pass the instance of this interface to this module? So that, the code here looked like
```
@popoverApi.show($("#formula"), options)
```

The reason is that we have our own code in Papeeria which is capable of showing popovers in the editor, and I'd like to reuse that code in production, while for demo purposes the code written here is okay.